### PR TITLE
Migrate all CORS requests from content context to background context

### PR DESCRIPTION
**Pull request by KarlCastle**
Continuing work on tfedor/Enhanced_Steam#62

### CODE REVIEW

File: js/background.js [hunk 1]
```diff
@@ -41,61 +38,482 @@ const LocalStorage = (function(){
...
+                    throw `Could not retrieve '${endpoint}'`;
+                delete json.result;
+                return json; // 'response': response, 
+            })
+        ;
+    };
+
     /**
-     * Requires user to be signed in, can we validate this from background?
+     * self.prices = async function({ 'params': params }) {
```
Comment (MxtOUT) [inline js/background.js]: Isn't this already substituted by the factory function?

Comment (KarlCastle) [reply js/background.js]: Err, yes. The comment was a note-to-self of what `AugmentedSteamApi.endpointFactory()` was supposed to return. It also helped as a guide as to "can I replace this function, or is it doing more than the guide?".

Comment (KarlCastle) [reply js/background.js]: Fixed in c173602

**Comment by KarlCastle** [comment]
This should now resolve #62

**Pull request merged**